CHANGES.rst

```diff
@@ -81,6 +89,26 @@ API changes
 
 - ``astropy.nddata``
 
+  - ``NDDataBase`` only defines an abstract ``uncertainty`` getter instead of
+    an actual implementation. The original was moved to ``NDData``.
+
+  - ``NDDataBase`` doesn't implement an ``uncertainty`` setter. It was moved to
+    ``NDData``. The setter was also modified:
+
+    - it wraps an ``uncertainty`` inside an ``UnknownUncertainty``
+      if no ``uncertainty_type`` attribute is present in the uncertainty instead
+      of raising an Exception.
+
+    - the uncertainty_type of the uncertainty must not be a string
+      anymore. But it is still recommended.
+
+    - did not set the ``parent_nddata`` of the ``uncertainty`` if the uncertainty
+      was a subclass of ``.NDUncertainty``. #4152
+
+  - ``NDData.__init__``: if the ``data`` is another instance or subclass of
```

Comment:
This one could be dropped I think. Doesn't really change the intended behavior, and it had not been reported as a bug.